feat: add `blas/ext/base/ndarray/gsorthp` (#7756)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/ndarray/gsorthp/docs/repl.txt

{{alias}}( arrays )
    Sorts a one-dimensional ndarray using heapsort.

>     The input strided array is sorted *in-place* (i.e., the input strided array
>     is *mutated*).

Reviewer: ```suggestion
    The input ndarray is sorted in-place (i.e., the input ndarray is mutated).
```

    Parameters
    ----------
    arrays: ArrayLikeObject<ndarray>
        Array-like object containing a one-dimensional input ndarray and a

>         zero-dimensional ndarray containing the sort order.

Reviewer: ```suggestion
        zero-dimensional ndarray specifying the sort order.
```

    Returns
    -------
    out: ndarray
        Input ndarray.

    Examples
    --------
    > var xbuf = [ 1.0, -2.0, 3.0, -4.0 ];
    > var dt = 'generic';
    > var sh = [ xbuf.length ];
    > var sx = [ 1 ];
    > var ox = 0;
    > var ord = 'row-major';
    > var x = new {{alias:@stdlib/ndarray/ctor}}( dt, xbuf, sh, sx, ox, ord );

>     > var o = {{alias:@stdlib/ndarray/from-scalar}}( 1.0, { 'dtype': dt } );

Author: ```suggestion
    > var o = {{alias:@stdlib/ndarray/from-scalar}}( 1.0 );
```
